# Upstream: 9e243df5b57e119a8ca4611761dd5f0c6716cbb6
# test-sandbox: Ignore PermissionError from xdg-document-portal
diff --git a/tests/test-sandbox.py b/tests/test-sandbox.py
index 9d5fd33..c34457e 100644
--- a/tests/test-sandbox.py
+++ b/tests/test-sandbox.py
@@ -137,7 +137,15 @@ class TestSandbox(unittest.TestCase):
                             f'expected bwrap to fail, stderr: {result.stderr!r}')
 
     def assertMountFlags(self, path, expected, unexpected):
-        flags = os.statvfs(path).f_flag
+        try:
+            flags = os.statvfs(path).f_flag
+        except PermissionError:
+            if path.endswith('/doc'):
+                # https://github.com/flatpak/xdg-desktop-portal/issues/553
+                return
+            else:
+                raise
+
         for f in expected:
             self.assertTrue(flags & f, f'{path}: expected flag {f:#x} not set')
         for f in unexpected:
@@ -314,7 +322,16 @@ class TestSandbox(unittest.TestCase):
             if any(mountpoint == p or mountpoint.startswith(p + '/')
                    for p in skip_prefixes):
                 continue
-            flags = os.statvfs(mountpoint).f_flag
+
+            try:
+                flags = os.statvfs(mountpoint).f_flag
+            except PermissionError:
+                if mountpoint.endswith('/doc'):
+                    # https://github.com/flatpak/xdg-desktop-portal/issues/553
+                    return
+                else:
+                    raise
+
             self.assertTrue(
                 flags & os.ST_RDONLY,
                 f'{mountpoint} is not read-only',
